src/test/java/com/teragrep/rlp_01/RelpParserTest.java

```diff
@@ -0,0 +1,163 @@
+        Assertions.assertEquals("close", parser.getCommandString(), "parser getCommandString() differs");
+    }
+
+    @Test
+    public void testCommandAbort() {
+        RelpParser parser = createParser("0 abort 0\n");
+        Assertions.assertTrue(parser.isComplete(), "parser isComplete() differs");
+        Assertions.assertEquals("abort", parser.getCommandString(), "parser getCommandString() differs");
+    }
+
+    @Test
+    public void testCommandServerClose() {
+        RelpParser parser = createParser("0 serverclose 0\n");
+        Assertions.assertTrue(parser.isComplete(), "parser isComplete() differs");
+        Assertions.assertEquals("serverclose", parser.getCommandString(), "parser getCommandString() differs");
+    }
+
+    @Test
+    public void testCommandSyslog() {
+        RelpParser parser = createParser("0 syslog 0\n");
+        Assertions.assertTrue(parser.isComplete(), "parser isComplete() differs");
+        Assertions.assertEquals("syslog", parser.getCommandString(), "parser getCommandString() differs");
+    }
+
+    @Test
+    public void testCommandResponse() {
+        RelpParser parser = createParser("0 rsp 0\n");
+        Assertions.assertTrue(parser.isComplete(), "parser isComplete() differs");
+        Assertions.assertEquals("rsp", parser.getCommandString(), "parser getCommandString() differs");
+    }
+
+    @Test
+    public void testReset() {
+        RelpParser parser = createParser("0 rsp 6 200 OK\n");
+        parser.reset();
+        Assertions.assertFalse(parser.isComplete(), "parser isComplete() differs");
+        Assertions.assertEquals(-1, parser.getTxnId(), "parser getTxnId() differs");
+        Assertions.assertEquals("", parser.getCommandString(), "parser getCommandString() differs");
+        Assertions.assertEquals(-1, parser.getLength(), "parser getLength() differs");
+        Assertions.assertNull(parser.getData(), "parser getData() differs");
+    }
+
+    @Test
+    public void testNonNumericTxnId() {
+        Assertions.assertThrows(NumberFormatException.class, () -> createParser("Seven rsp 6 200 OK\n"));
+    }
+
+    @Test
+    public void testNonNumericLength() {
+        Assertions.assertThrows(NumberFormatException.class, () -> createParser("0 rsp six 200 OK\n"));
+    }
+
+    @Test
+    public void testTooLongMessage() {
+        Assertions.assertThrows(IllegalStateException.class, () -> createParser("0 rsp 6 My Message Is Too Long"));
+    }
+
+    @Test
+    public void testMultipleLastNewlines() {
+        RelpParser parser = createParser("0 rsp 3 six\n\n\n\n\n\n");
+        Assertions.assertEquals("six", StandardCharsets.UTF_8.decode(parser.getData()).toString(), "parser getData() differs");
+        Assertions.assertTrue(parser.isComplete(), "parser isComplete() differs");
+    }
+
+    @Test
+    public void testDataAfterLastNewline() {
+        Assertions.assertThrows(IllegalStateException.class, () -> createParser("0 rsp 3 six\n\nBonjour"));
```

Comment:
this is valid because six is 3 bytes long, see previous

Reply:
Removed the test